packages/assets/src/loader/parsers/loadBitmapFont.ts

```diff
@@ -41,35 +22,37 @@ export const loadBitmapFont = {
 
     async testParse(data: string): Promise<boolean>
     {
-        const isText = TextFormat.test(data);
-        const isXMLText = XMLStringFormat.test(data);
-
-        return isText || isXMLText;
+        return TextFormat.test(data) || XMLStringFormat.test(data);
     },
 
     async parse(asset: string, data: LoadAsset, loader: Loader): Promise<BitmapFont>
     {
-        const isText = TextFormat.test(asset);
+        const fontData: BitmapFontData = TextFormat.test(asset)
+            ? TextFormat.parse(asset)
+            : XMLStringFormat.parse(asset);
+
+        const { src } = data;
+        const { page: pages } = fontData;
+        const textureUrls = [];
 
-        if (isText)
+        for (let i = 0; i < pages.length; ++i)
         {
-            const parsed = TextFormat.parse(asset);
+            const pageFile = pages[i].file;
+            const imagePath = join(dirname(src), pageFile);
 
-            return await _loadBitmap(data.src, parsed, loader);
+            textureUrls.push(imagePath);
         }
 
-        return await _loadBitmap(data.src, XMLStringFormat.parse(asset), loader);
+        const textures: Texture[] = Object.values(await loader.load(textureUrls));
+
+        return BitmapFont.install(fontData, textures, true);
     },
 
-    async load(url: string, _options: LoadAsset, loader: Loader): Promise<BitmapFont>
+    async load(url: string, _options: LoadAsset): Promise<string>
     {
         const response = await settings.ADAPTER.fetch(url);
 
-        const text = await response.text();
-
-        const data = new window.DOMParser().parseFromString(text, 'text/xml');
-
-        return await _loadBitmap(url, XMLFormat.parse(data), loader);
+        return response.text();
```

Comment:
If someone is trying to load an XML file that is _not_ a font, this means they would get back an XML string instead of an XMLDocument. We may need a generic loader like the json/txt loaders. Using XML for data isn't super common these days, so I'm not that concerned about it.

Reply:
I don't fully remember how the unloading works but returning a non `BitmapFont` might cause an issue as we call `font.destroy()` in `unload`

I will test this out and report back